apps/server/src/modules/group/repo/group.repo.ts

```diff
@@ -98,4 +155,65 @@ export class GroupRepo extends BaseDomainObjectRepo<Group, GroupEntity> {
 
 		return domainObjects;
 	}
+
+	private async findAvailableGroup(
+		pipelineStage: unknown[],
+		skip = 0,
+		limit?: number,
+		nameQuery?: string
+	): Promise<Page<Group>> {
+		let nameRegexFilter = {};
+		const pipeline: unknown[] = pipelineStage;
```

Comment:
the js assignment operation is not a copy operation. you just assign the reference of `pipelineStage` to `pipeline` making `pipeline` just an alias to `pipelineStage` btw. (so it has side effects)